kodilla-testing/src/test/java/com/kodilla/testing/library/BookDirectoryTestSuite.java

```diff
@@ -0,0 +1,140 @@
+        // Then
+
+        assertEquals(0, theListOfBooks0.size());
+        assertEquals(15, theListOfBooks15.size());
+        assertEquals(0, theListOfBooks40.size());
+    }
+
+    @Test
+    public void testListBooksWithConditionFragmentShorterThan3() {
+        // Given
+        LibraryDatabase libraryDatabaseMock = mock(LibraryDatabase.class);
+        BookLibrary bookLibrary = new BookLibrary(libraryDatabaseMock);
+        List<Book> resultListOf10Books = generateListOfNBooks(10);
+        when(libraryDatabaseMock.listBooksWithCondition(anyString()))
+                .thenReturn(resultListOf10Books);
+
+        // When
+        List<Book> theListOfBooks10 = bookLibrary.listBooksWithCondition("An");
+
+        // Then
+        assertEquals(0, theListOfBooks10.size());
+        verify(libraryDatabaseMock, times(0)).listBooksWithCondition(anyString());
+    }
+
+    @Test
+    public void testListBooksInHandsOf0Books(){
+        // Given
+        LibraryDatabase libraryDatabaseMock = mock(LibraryDatabase.class);
+        BookLibrary bookLibrary = new BookLibrary(libraryDatabaseMock);
+        LibraryUser libraryUser1 = new LibraryUser("Thomas", "Stone","89071800000");
+        List<Book> resultListOf0Books = new ArrayList<>(); // = generateListOfNBooks(0);
+        when(libraryDatabaseMock.listBooksInHandsOf(libraryUser1)).thenReturn(resultListOf0Books);
+
+        // When
+        List<Book> theListLibraryUser1 = bookLibrary.listBooksInHandsOf(libraryUser1);
+
+        // Then
+        assertEquals(0, theListLibraryUser1.size());
+
+    }
+
+    @Test
+    public void testListBooksInHandsOf1Books(){
+        // Given
+        LibraryDatabase libraryDatabaseMock = mock(LibraryDatabase.class);
+        BookLibrary bookLibrary = new BookLibrary(libraryDatabaseMock);
+        LibraryUser libraryUser1 = new LibraryUser("Thomas", "Stone","89071800000");
+        List<Book> resultListOf1Books = generateListOfNBooks(1);
+        when(libraryDatabaseMock.listBooksInHandsOf(libraryUser1)).thenReturn(resultListOf1Books);
+
+        // When
+        List<Book> theListLibraryUser1 = bookLibrary.listBooksInHandsOf(libraryUser1);
+
+        // Then
+        assertEquals(1, theListLibraryUser1.size());
+
+    }
+
+    @Test
+    public void testListBooksInHandsOf5Books(){
+        // Given
+        LibraryDatabase libraryDatabaseMock = mock(LibraryDatabase.class);
+        BookLibrary bookLibrary = new BookLibrary(libraryDatabaseMock);
+        LibraryUser libraryUser1 = new LibraryUser("Thomas", "Stone","89071800000");
+        List<Book> resultListOf5Books = generateListOfNBooks(5);
+        when(libraryDatabaseMock.listBooksInHandsOf(libraryUser1)).thenReturn(resultListOf5Books);
+
+        // When
+        List<Book> theListLibraryUser1 = bookLibrary.listBooksInHandsOf(libraryUser1);
+
+        // Then
+        assertEquals(5, theListLibraryUser1.size());
+    }
+}
```

Comment:
powtarzajace sie elementy jak
LibraryDatabase libraryDatabaseMock = mock(LibraryDatabase.class);
dobrze byloby wyrzucic do zmiennych klasowych zamiast kopiowac do kazdego testu z osobna

Reply:
Nie za bardzo wiem jak to zrobić :/ a w przykładzie realizacji testów metody : listBooksWithCondition(String titleFragment) nie robili tak.